src/App.js

```diff
@@ -14,11 +15,103 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: [],
+  };
+
+  componentDidMount() {
+    this.setState({
+      selectedGoods: ['Jam'],
+    });
+  }
+
+  addGood = (good) => {
+    this.setState(({ selectedGoods }) => ({
+      selectedGoods: [
+        ...selectedGoods,
+        good,
+      ],
+    }));
+  }
+
+  removeGood = (good) => {
+    this.setState(({ selectedGoods }) => ({
+      selectedGoods: selectedGoods.filter(
+        selectedGood => selectedGood !== good,
+      ),
+    }));
+  }
+
+  clearSelection = () => {
+    this.setState({
+      selectedGoods: [],
+    });
+  };
+
+  renderButton = (className, clickHandler, label) => (
+    <button
+      type="button"
+      className={ClassNames('btn', className)}
+      onClick={clickHandler}
+    >
+      {label}
+    </button>
+  )
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <header className="App__header">
+          {
+            selectedGoods.length !== 0 && this.renderButton(
+              'App__button App__button--clear',
+              this.clearSelection.bind(this),
+              'X',
+            )
+          }
+          <h1 className="App__title">
+            {
+              selectedGoods.length !== 0
+                ? `Selected goods: ${selectedGoods}`
+                : 'No goods selected'
+            }
+          </h1>
+        </header>
+        <ul className="GoodsList">
+          {goodsFromServer.map(good => (
+            <li
+              key={good}
+              className={ClassNames(
+                'GoodsList__item',
+                selectedGoods.includes(good) && 'GoodsList__item--selected',
```

Comment:
```suggestion
                { 'GoodsList__item--selected': selectedGoods.includes(good) }
```
може зберігати в об'єкт, а не в масив? тоді можна буде робити доступ по ключу зі складністю O(1)